Maven Dependency update is failing when newly added component/kamelet/processor is missing some mandatory parameters

```
Wed, 24 Sep 2025 08:59:28 GMT] [ERROR] org.apache.camel.RuntimeCamelException: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: aws-bedrock-agent-runtime-sink
	at org.apache.camel.RuntimeCamelException.wrapRuntimeException(RuntimeCamelException.java:68)
	at org.apache.camel.support.service.BaseService.doFail(BaseService.java:431)
	at org.apache.camel.impl.engine.AbstractCamelContext.doFail(AbstractCamelContext.java:3257)
	at org.apache.camel.support.service.BaseService.fail(BaseService.java:360)
	at org.apache.camel.impl.engine.AbstractCamelContext.failOnStartup(AbstractCamelContext.java:4289)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2220)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2226)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:211)
	at org.apache.camel.main.KameletMain.doStart(KameletMain.java:400)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:123)
	at org.apache.camel.dsl.jbang.core.commands.Run.runKameletMain(Run.java:1633)
	at org.apache.camel.dsl.jbang.core.commands.Run.run(Run.java:1002)
	at org.apache.camel.dsl.jbang.core.commands.Run.runExport(Run.java:380)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.runSilently(ExportBaseCommand.java:387)
	at org.apache.camel.dsl.jbang.core.commands.ExportQuarkus.export(ExportQuarkus.java:75)
	at org.apache.camel.dsl.jbang.core.commands.Export.export(Export.java:230)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doExport(DependencyList.java:267)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.export(DependencyList.java:84)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.doCall(ExportBaseCommand.java:304)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doCall(DependencyList.java:62)
	at org.apache.camel.dsl.jbang.core.commands.DependencyUpdate.doCall(DependencyUpdate.java:89)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:71)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:37)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:205)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:72)
	at main.CamelJBang.main(CamelJBang.java:36)
Caused by: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: aws-bedrock-agent-runtime-sink
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:510)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2590)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:85)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2207)
	... 29 more
Caused by: org.apache.camel.component.kamelet.FailedToCreateKameletException: Error creating or loading Kamelet with id aws-bedrock-agent-runtime-sink (locations: classpath:kamelets,github:apache:camel-kamelets/kamelets)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:498)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.createRouteForEndpoint(KameletComponent.java:462)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:507)
	... 32 more
Caused by: java.lang.IllegalArgumentException: Route template aws-bedrock-agent-runtime-sink the following mandatory parameters must be provided: modelId, knowledgeBaseId, region
	at org.apache.camel.impl.DefaultModel.doAddRouteFromTemplate(DefaultModel.java:508)
	at org.apache.camel.impl.DefaultModel.addRouteFromKamelet(DefaultModel.java:451)
	at org.apache.camel.impl.DefaultCamelContext.addRouteFromKamelet(DefaultCamelContext.java:391)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:485)
	... 34 more
```

```
[Wed, 24 Sep 2025 09:02:14 GMT] [ERROR] org.apache.camel.RuntimeCamelException: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: azure-cosmosdb-sink
	at org.apache.camel.RuntimeCamelException.wrapRuntimeException(RuntimeCamelException.java:68)
	at org.apache.camel.support.service.BaseService.doFail(BaseService.java:431)
	at org.apache.camel.impl.engine.AbstractCamelContext.doFail(AbstractCamelContext.java:3257)
	at org.apache.camel.support.service.BaseService.fail(BaseService.java:360)
	at org.apache.camel.impl.engine.AbstractCamelContext.failOnStartup(AbstractCamelContext.java:4289)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2220)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2226)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:211)
	at org.apache.camel.main.KameletMain.doStart(KameletMain.java:400)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:123)
	at org.apache.camel.dsl.jbang.core.commands.Run.runKameletMain(Run.java:1633)
	at org.apache.camel.dsl.jbang.core.commands.Run.run(Run.java:1002)
	at org.apache.camel.dsl.jbang.core.commands.Run.runExport(Run.java:380)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.runSilently(ExportBaseCommand.java:387)
	at org.apache.camel.dsl.jbang.core.commands.ExportQuarkus.export(ExportQuarkus.java:75)
	at org.apache.camel.dsl.jbang.core.commands.Export.export(Export.java:230)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doExport(DependencyList.java:267)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.export(DependencyList.java:84)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.doCall(ExportBaseCommand.java:304)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doCall(DependencyList.java:62)
	at org.apache.camel.dsl.jbang.core.commands.DependencyUpdate.doCall(DependencyUpdate.java:89)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:71)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:37)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:205)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:72)
	at main.CamelJBang.main(CamelJBang.java:36)
Caused by: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: azure-cosmosdb-sink
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:510)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2590)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:85)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2207)
	... 29 more
Caused by: org.apache.camel.component.kamelet.FailedToCreateKameletException: Error creating or loading Kamelet with id azure-cosmosdb-sink (locations: classpath:kamelets,github:apache:camel-kamelets/kamelets)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:498)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.createRouteForEndpoint(KameletComponent.java:462)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:507)
	... 32 more
Caused by: java.lang.IllegalArgumentException: Route template azure-cosmosdb-sink the following mandatory parameters must be provided: databaseName, containerName, databaseEndpoint
	at org.apache.camel.impl.DefaultModel.doAddRouteFromTemplate(DefaultModel.java:508)
	at org.apache.camel.impl.DefaultModel.addRouteFromKamelet(DefaultModel.java:451)
	at org.apache.camel.impl.DefaultCamelContext.addRouteFromKamelet(DefaultCamelContext.java:391)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:485)
	... 34 more
```

```
[Wed, 24 Sep 2025 09:04:00 GMT] [ERROR] org.apache.camel.RuntimeCamelException: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: exec-sink
	at org.apache.camel.RuntimeCamelException.wrapRuntimeException(RuntimeCamelException.java:68)
	at org.apache.camel.support.service.BaseService.doFail(BaseService.java:431)
	at org.apache.camel.impl.engine.AbstractCamelContext.doFail(AbstractCamelContext.java:3257)
	at org.apache.camel.support.service.BaseService.fail(BaseService.java:360)
	at org.apache.camel.impl.engine.AbstractCamelContext.failOnStartup(AbstractCamelContext.java:4289)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2220)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2226)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:211)
	at org.apache.camel.main.KameletMain.doStart(KameletMain.java:400)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:123)
	at org.apache.camel.dsl.jbang.core.commands.Run.runKameletMain(Run.java:1633)
	at org.apache.camel.dsl.jbang.core.commands.Run.run(Run.java:1002)
	at org.apache.camel.dsl.jbang.core.commands.Run.runExport(Run.java:380)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.runSilently(ExportBaseCommand.java:387)
	at org.apache.camel.dsl.jbang.core.commands.ExportQuarkus.export(ExportQuarkus.java:75)
	at org.apache.camel.dsl.jbang.core.commands.Export.export(Export.java:230)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doExport(DependencyList.java:267)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.export(DependencyList.java:84)
	at org.apache.camel.dsl.jbang.core.commands.ExportBaseCommand.doCall(ExportBaseCommand.java:304)
	at org.apache.camel.dsl.jbang.core.commands.DependencyList.doCall(DependencyList.java:62)
	at org.apache.camel.dsl.jbang.core.commands.DependencyUpdate.doCall(DependencyUpdate.java:89)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:71)
	at org.apache.camel.dsl.jbang.core.commands.CamelCommand.call(CamelCommand.java:37)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:205)
	at org.apache.camel.dsl.jbang.core.commands.CamelJBangMain.run(CamelJBangMain.java:72)
	at main.CamelJBang.main(CamelJBang.java:36)
Caused by: org.apache.camel.VetoCamelContextStartException: Failure creating route from template: exec-sink
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:510)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2590)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:85)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2207)
	... 29 more
Caused by: org.apache.camel.component.kamelet.FailedToCreateKameletException: Error creating or loading Kamelet with id exec-sink (locations: classpath:kamelets,github:apache:camel-kamelets/kamelets)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:498)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.createRouteForEndpoint(KameletComponent.java:462)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:507)
	... 32 more
Caused by: java.lang.IllegalArgumentException: Route template exec-sink the following mandatory parameters must be provided: executable
	at org.apache.camel.impl.DefaultModel.doAddRouteFromTemplate(DefaultModel.java:508)
	at org.apache.camel.impl.DefaultModel.addRouteFromKamelet(DefaultModel.java:451)
	at org.apache.camel.impl.DefaultCamelContext.addRouteFromKamelet(DefaultCamelContext.java:391)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.doCreateRouteForEndpoint(KameletComponent.java:485)
	... 34 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 # 2.8.0
 
+- Allow to update Maven dependencies on not fully configured route
 - Use Node 22 instead of Node 20
 - show `What's New` release page after each extension upgrade (only for first startup)
 

--- a/src/helpers/CamelJBang.ts
+++ b/src/helpers/CamelJBang.ts
@@ -96,7 +96,16 @@ export class CamelJBang {
 	 */
 	public async dependencyUpdate(pomPath: string, routePath: string, cwd?: string): Promise<number> {
 		return await new Promise<number>((resolve) => {
-			const args: string[] = [`-Dcamel.jbang.version=${this.camelJBangVersion}`, 'camel@apache/camel', 'dependency', 'update', pomPath, routePath];
+			const args: string[] = [
+				`-Dcamel.jbang.version=${this.camelJBangVersion}`,
+				'camel@apache/camel',
+				'dependency',
+				'update',
+				pomPath,
+				routePath,
+				'--lazy-bean',
+				'--ignore-loading-error',
+			];
 			KaotoOutputChannel.logInfo(`Camel Dependency Update: jbang ${args.join(' ')}`);
 			execFile(this.jbang, args, { cwd, maxBuffer: 50 * 1024 * 1024 }, (error, stdout, stderr) => {
 				if (stdout) {

